Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -245,7 +246,10 @@ extension BrowserViewController: WKNavigationDelegate {
         info["url"] = webView.URL
         info["title"] = webView.title
 
-        notificationCenter.postNotificationName("LocationChange", object: self, userInfo: info)
+        webView.runScript("getAll()", js: "Favicons", callback: { obj in
```

Comment:
It could become a `FaviconHelper` that finds FavIcons upon page load and then uses the WKUserScriptMessage API to push that back to the FaviconHelper which can then post a notification or call a delegate to notify 'the thing that deals with icons`.